### DISCUSSION TITLE
High CPU Usuage

### DISCUSSION THREAD

**Discussion by uwe72** [ultralytics/yolov5#2289] [Q&A]
I were able to start the detect.py in order to do the detections of my IP-Camera using a mpeg/rtsp stream. Works well:
![image](https://user-images.githubusercontent.com/61848539/109028588-3b90a400-76c2-11eb-8aad-bb6eb574b4ed.png)

My problem is that the CPU is arround 60%.

I am using a NUC10i7FNH. Is there a way to decrease the CPU?

**Comment by glenn-jocher** [comment]
@uwe72 use a GPU?

**Reply by uwe72** [reply]
no, i don't use GPU

**Reply by glenn-jocher** [reply]
@uwe72 yes exactly. If you want to offload inference from your CPU you should use a GPU.

**Reply by uwe72** [reply]
@glenn-jocher 
Thank you!

I am brave now and ask :-) sorry for this probably stupid beginner question: Has every hardware, or in my case my Intel NUC a GPU? If not, can i add or exchange anything on the hardware?

Second question: If the hardware has a GPU how can i start the software in docker in the GPU mode?

**Reply by glenn-jocher** [reply]
Supported GPUs are here: https://developer.nvidia.com/cuda-gpus#compute

For Docker see Docker quickstart guide below:

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/GCP-Quickstart)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/AWS-Quickstart)
- **Docker Image**. See [Docker Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/Docker-Quickstart) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu.

**Reply by uwe72** [reply]
Thanks!!!
